### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by nrwiersma**

### CODE REVIEW

File: format.go [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+	case int:
+		buf.AppendInt(int64(v))
+	case int8:
+		buf.AppendInt(int64(v))
+	case int16:
+		buf.AppendInt(int64(v))
+	case int32:
+		buf.AppendInt(int64(v))
+	case int64:
+		buf.AppendInt(v)
```
Comment (michalkurzeja) [inline format.go:95]: All the int cases could be merged into 1. Why not do it and avoid repetition?

Comment (michalkurzeja) [reply format.go:95]: Same for uints and floats.

File: format.go [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+	}
+
+	switch v := value.(type) {
+	case time.Time:
+		buf.AppendTime(v, timeFormat)
+	case bool:
+		buf.AppendBool(v)
+	case float32:
+		buf.AppendFloat(float64(v), 'f', 3, 64)
+	case float64:
```
Comment (michalkurzeja) [inline format.go:162]: Same concern about repeated cases.

File: handler.go [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+		w.Write(fmtr.Format(msg, lvl, ctx))
+		mu.Unlock()
+	}
+
+	return HandlerFunc(h)
+}
+
+// FilterHandler returns a handler that only writes messages to the wrapped
+// handler if the given function evaluates true.
+func FilterHandler(fn func(msg string, lvl Level, ctx []interface{}) bool, h Handler) Handler {
```
Comment (michalkurzeja) [inline handler.go]: Make a FilterFunc type for a more readable signature of FilterHandler??

File: logger_test.go [hunk 4]
```diff
@@ -0,0 +1,161 @@
+package logged_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
```
Comment (michalkurzeja) [inline logger_test.go]: Fix fmt

**Review by michalkurzeja — CHANGES_REQUESTED** [review]

File: format.go (hunk 1 above)
Comment (nrwiersma) [reply format.go:95]: How is that?

**Review by michalkurzeja — APPROVED** [review]

**Pull request merged**